botorch/acquisition/hentropy_search.py

```diff
@@ -0,0 +1,282 @@
+
+        Returns:
+            The augmented size for optimization.
+        """
+
+        return q + self.cfg.n_dim_action
+
+    def extract_candidates(self, batch_as_full: Tensor) -> Tensor:
+        r"""We only return X as the set of candidates post-optimization.
+
+        Args:
+            batch_as_full: A `b x (q + num_fantasies) x d`-dim Tensor with `b`
+                t-batches of `q + num_fantasies` design points each.
+
+        Returns:
+            A `b x q x d`-dim Tensor with `b` t-batches of `q` design points each.
+        """
+
+        assert len(batch_as_full.shape) == 2
+        n_restarts = batch_as_full.size(0)
+        split_sizes = [
+            batch_as_full.size(1) - self.cfg.n_actions * self.cfg.n_dim_action,
+            self.cfg.n_actions * self.cfg.n_dim_action,
+        ]
+        batch_xs, batch_as_full = torch.split(batch_as_full, split_sizes, dim=1)
+        batch_xs = batch_xs.reshape(n_restarts, -1, self.cfg.n_dim_design)
+        batch_as_full = batch_as_full.reshape(
+            n_restarts, self.cfg.n_actions, self.cfg.n_dim_action
+        )
+
+        return batch_xs
+
+
+class qLossFunctionTopK(nn.Module):
+    r"""Batch loss function for the task of finding top-K."""
+
+    def __init__(self, dist_weight=1.0, dist_threshold=0.5) -> None:
+        r"""Batch loss function for the task of finding top-K.
+
+        Args:
+            loss_function_hyperparameters: hyperparameters for the loss function class.
+        """
+
+        super().__init__()
+        self.register_buffer("dist_weight", torch.as_tensor(dist_weight))
+        self.register_buffer(
+            "dist_threshold",
+            torch.as_tensor(dist_threshold),
+        )
+
+    def forward(self, A: Tensor, Y: Tensor) -> Tensor:
+        r"""Evaluate batch loss function on a tensor of actions.
+
+        Args:
+            A: Actor tensor with shape `batch_size x n_fantasy_at_design_pts
+                x num_actions x action_dim`.
+            Y: Fantasized sample with shape `n_fantasy_at_action_pts x
+                n_fantasy_at_design_pts x batch_size x num_actions`.
+
+        Returns:
+            A Tensor of shape `n_fantasy_at_action_pts x batch`.
+        """
+
+        Y = Y.sum(dim=-1).mean(dim=0)
+        # >>> n_fantasy_at_design_pts x batch_size
+
+        num_actions = A.shape[-2]
+
+        dist_reward = 0
+        if num_actions >= 2:
+            A = A.contiguous()
+            A_distance = torch.cdist(A, A, p=1.0)
+            A_distance_triu = torch.triu(A_distance)
+            # >>> n_fantasy_at_design_pts x batch_size x num_actions x num_actions
+
+            A_distance_triu[A_distance_triu > self.dist_threshold] = self.dist_threshold
+            denominator = num_actions * (num_actions - 1) / 2.0
+            dist_reward = A_distance_triu.sum((-1, -2)) / denominator
+            # >>> n_fantasy_at_design_pts x batch_size
+
+        q_hes = Y + self.dist_weight * dist_reward
```

Comment:
As written, I think this will error if Y is on the GPU, since `self.dist_weight` is a CPU tensor. (Does `dist_weight` need to be a tensor at all?)

```suggestion
        q_hes = Y + self.dist_weight.to(Y.device) * dist_reward
```

Reply:
looks like `dist_weight` is not a tensor anymore (though docstring needs updating). If that's always the case this comment should be resolved.